packages/memory-inspector/src/browser/editable-widget/memory-editable-table-widget.tsx

```diff
@@ -0,0 +1,324 @@
+            const { value } = this.composeByte(address, true);
+            const { value: inMemory } = this.composeByte(address, false);
+            const oldValue = this.previousBytes && this.composeByte(address, false, this.previousBytes).value;
+            const decimal = parseInt(value, 16);
+            const octal = decimal.toString(8).padStart(this.options.byteSize / 8, '0');
+            const UTF8 = String.fromCharCode(decimal);
+            const binary = this.getPaddedBinary(decimal);
+            const toSend: EasilyMappedObject = { hex: value, octal, binary, decimal };
+            if (UTF8) {
+                toSend.UTF8 = UTF8;
+            }
+            if (inMemory !== value) {
+                toSend['Current Value'] = inMemory;
+            }
+            if (oldValue !== undefined && oldValue !== value) {
+                toSend['Previous Value'] = oldValue;
+            }
+            return toSend;
+        }
+        return undefined;
+    }
+
+    protected composeByte(
+        addressPlusArrayOffset: Long,
+        usePendingEdits: boolean,
+        dataSource: Uint8Array = this.memory.bytes,
+    ): Interfaces.ByteFromChunkData {
+        let value = '';
+
+        const offset = addressPlusArrayOffset.subtract(this.memory.address);
+        const chunksPerByte = this.options.byteSize / 8;
+        const startingChunkIndex = offset.subtract(offset.modulo(chunksPerByte));
+        const address = this.memory.address.add(startingChunkIndex.divide(chunksPerByte));
+        for (let i = 0; i < chunksPerByte; i += 1) {
+            const targetOffset = startingChunkIndex.add(i);
+            const targetChunk = this.getFromMapOrArray(targetOffset, usePendingEdits, dataSource);
+            value += targetChunk.padStart(2, '0');
+        }
+
+        return { address, value };
+    }
+
+    protected getFromMapOrArray(arrayOffset: Long, usePendingEdits: boolean, dataSource: Uint8Array = this.memory.bytes): string {
+        let value = usePendingEdits ? this.pendingMemoryEdits.get(arrayOffset.add(this.memory.address).toString()) : undefined;
+        if (value === undefined) {
+            value = dataSource[arrayOffset.toInt()]?.toString(16) ?? '';
+        }
+        return value;
+    }
+
+    protected handleClearEditClick = (): void => this.clearEdits();
+
+    protected clearEdits(address?: Long): void {
+        if (typeof address === 'number') {
+            this.pendingMemoryEdits.delete(address);
+        } else {
+            this.pendingMemoryEdits.clear();
+        }
+        this.update();
+    }
+
+    protected submitMemoryEdits = async (): Promise<void> => {
+        this.memoryEditsCompleted = new Deferred();
+        for (const edit of this.createUniqueEdits()) {
+            try {
+                await this.memoryProvider.writeMemory(edit);
+            } catch (e) {
+                console.log('Problem writing memory with arguments', edit, '\n', e);
```

Comment:
I think it would be great if we could show the errors below the pending edits within the table footer - similar to how read errors are shown in the options widget. What do you think?

Reply:
Done.